app/User.php

```diff
@@ -453,6 +462,35 @@ public function isFrauding(): bool
             ->count() >= 2;
     }
 
+    public function requiresEmail(): self
```

Comment:
This 30 line function is now loaded in memory on every single page load of Kanka, but is used in a single place. Please refactor to put it in a service.